environments/jvm/src/main/java/io/fission/Server.java

```diff
@@ -0,0 +1,136 @@
+        long startTime = System.nanoTime();
+    	File file = new File(req.getFilepath());
+    	if (!file.exists()) {
+    		return ResponseEntity.badRequest().body("/userfunc/user not found");
+    	}
+    	
+    	String entryPoint = req.getFunctionName();
+    	System.out.println("Entrypoint class:"+ entryPoint);
+    	if (entryPoint == null) {
+    		return ResponseEntity.badRequest().body("Entrypoint class is missing in the function");
+    	}
+
+    	JarFile jarFile = null;
+    	ClassLoader cl = null;
+    	try {
+    		
+    		jarFile = new JarFile(file);
+    		Enumeration<JarEntry> e = jarFile.entries();
+    		URL[] urls = { new URL("jar:file:" + file+"!/") };
+
+    		//TODO Check if the classloading can be improved for ex. use something like: 
+    		// Thread.currentThread().setContextClassLoader(cl);
+    		if (this.getClass().getClassLoader() == null) {
+    			cl = URLClassLoader.newInstance(urls);	
+    		} else {
+    			cl = URLClassLoader.newInstance(urls, this.getClass().getClassLoader());
+    		}
+    		
+    		if (cl == null) {
+    			return ResponseEntity.status(500).body("Failed to initialize the classloader");
+    		}
+    		
+    		// Load all dependent classes from libraries etc. 
+    		while (e.hasMoreElements()) {
+    		    JarEntry je = e.nextElement();
+    		    if(je.isDirectory() || !je.getName().endsWith(".class")){
+    		        continue;
+    		    }
+    		    String className = je.getName().substring(0,je.getName().length()-6);
+    		    className = className.replace('/', '.');
+    		    System.out.println("Class="+ className);
+    		    cl.loadClass(className);
+    		}
+    		
+    		// Instantiate the function class
+			fn = (FissionFunction<RequestEntity, FissionContext>) cl.loadClass(entryPoint).newInstance();
+			
+		} catch (MalformedURLException e) {
+			e.printStackTrace();
+			return ResponseEntity.badRequest().body("Error loading the Function class file");
+		} catch (ClassNotFoundException e) {
+			e.printStackTrace();
+			return ResponseEntity.badRequest().body("Error loading Function or dependent class");
+		} catch (InstantiationException e) {
+			e.printStackTrace();
+			return ResponseEntity.badRequest().body("Error creating a new instance of function class");
+		} catch (IllegalAccessException e) {
+			e.printStackTrace();
+			return ResponseEntity.badRequest().body("Error creating a new instance of function class");
+		} catch (IOException e) {
+			e.printStackTrace();
+			return ResponseEntity.badRequest().body("Error reading the JAR file");
+		} finally {
+			try {
+				//cl.close();
+				jarFile.close();
+			} catch (IOException e) {
+				e.printStackTrace();
+				return ResponseEntity.badRequest().body("Error closing the file while loading the class");
+			}
+		}
+    	long elapsedTime = System.nanoTime() - startTime;
+    	System.out.println("Specialize call done in: " + elapsedTime/1000000 +" ms");
```

Comment:
@erwinvaneyk Checking the specialization time in the executor is not ideal, even more so for new deployment executor as the specialization call happens in a different flow.

Reply:
For logging - using a library would mean environment JVM has already loaded the library. This can cause a lot of issues when users import their own logging libraries etc. So I am trying to avoid using a logging library. I will see if Java native logger is more performant and use that

Reply:
Specialization issue: 👍 

Logging: you can use `java.util.Logging` which is included in the standard library. Or, you could use SLF4J which decouples the logging from the underlying logging library, allowing users to choose their own logger.